vnext/Microsoft.ReactNative/Fabric/Composition/Modal/WindowsModalHostViewComponentView.cpp

```diff
@@ -11,34 +10,177 @@
+}
+
+// constants for creating a new windows (code mostly taken from LogBox)
+constexpr PCWSTR c_modalWindowClassName = L"MS_REACTNATIVE_MODAL";
+constexpr auto CompHostProperty = L"CompHost";
+const int MODAL_DEFAULT_WIDTH = 500;
+const int MODAL_DEFAULT_HEIGHT = 500;
+
+// creates a new modal window
+void WindowsModalHostComponentView::ShowOnUIThread() {
+  auto host =
+      winrt::Microsoft::ReactNative::implementation::ReactNativeHost::GetReactNativeHost(m_context.Properties());
+  if (!host) {
+    return;
+  }
+
+  RegisterWndClass(); // creates and register a windows class
+
+  if (!m_hwnd) {
+    auto CompositionHwndHost = winrt::Microsoft::ReactNative::CompositionHwndHost();
+    winrt::Microsoft::ReactNative::ReactViewOptions viewOptions;
+    viewOptions.ComponentName(L"Modal");
+    CompositionHwndHost.ReactViewHost(
+        winrt::Microsoft::ReactNative::ReactCoreInjection::MakeViewHost(host, viewOptions));
+    HINSTANCE hInstance = GetModuleHandle(NULL);
+    winrt::impl::abi<winrt::Microsoft::ReactNative::ICompositionHwndHost>::type *pHost{nullptr};
+    winrt::com_ptr<::IUnknown> spunk;
+    CompositionHwndHost.as(spunk);
+
+    // get the root hwnd
+    auto roothwnd = reinterpret_cast<HWND>(
+        winrt::Microsoft::ReactNative::ReactCoreInjection::GetTopLevelWindowId(m_context.Properties().Handle()));
+
+    m_hwnd = CreateWindow(
+        c_modalWindowClassName,
+        L"React-Native Modal",
+        WS_OVERLAPPEDWINDOW,
+        CW_USEDEFAULT,
+        CW_USEDEFAULT,
+        MODAL_DEFAULT_WIDTH,
+        MODAL_DEFAULT_HEIGHT,
+        roothwnd, // parent
+        nullptr,
+        hInstance,
+        spunk.get());
+
+    // Check if window creation succeeded
+    if (!m_hwnd) {
+      throw std::exception("Failed to create new hwnd for Modal: " + GetLastError());
+    }
+
+    spunk.detach();
+  }
+
+  ShowWindow(m_hwnd, SW_NORMAL);
+  BringWindowToTop(m_hwnd);
+  SetFocus(m_hwnd);
+}
+
+void WindowsModalHostComponentView::HideOnUIThread() noexcept {
+  if (m_hwnd) {
+    ::ShowWindow(m_hwnd, SW_HIDE);
+  }
+}
+
+// Windows Procedure - callback function used for handling all messages (generated by NTUser or manual calls to
+// SendMessage)
+LRESULT CALLBACK ModalBoxWndProc(HWND hwnd, UINT message, WPARAM wparam, LPARAM lparam) noexcept {
+  auto data = reinterpret_cast<::IUnknown *>(GetProp(
+      hwnd,
+      CompHostProperty)); // gets data handle from the property list of specified window (ie the window we want to make)
+  winrt::Microsoft::ReactNative::CompositionHwndHost host{nullptr};
+
+  if (data) {
+    winrt::check_hresult(data->QueryInterface(
+        winrt::guid_of<winrt::Microsoft::ReactNative::CompositionHwndHost>(),
+        winrt::put_abi(host))); // look into the data for a CompositionHwndHost and store it in host
+    auto result = static_cast<LRESULT>(host.TranslateMessage(message, wparam, lparam));
+    if (result) {
+      return result;
+    }
```

Comment:
The classic Windows [TranslateMessage](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-translatemessage) returns a BOOL, I'm not sure why our host returns an LRESULT type. I guess because it actually calls [SendMessage](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-sendmessage). This is a confusing rename... with the burden of those return types confusingly passed on here.

But since it does return an LRESULT... you shouldn't need the cast?
```suggestion
    auto result = host.TranslateMessage(message, wparam, lparam);
    if (result) {
      return result;
    }
```

Reply:
type in the .idl file is a Int64 while code specifies a LRESULT, which needs a cast :/

Reply:
TranslateMessage either calls SendMessage or returns 0. The RootView handles that SendMessage and either calls into the event handler or returns 0. As far as I can tell, every single path in the event handler returns 0.
@acoates-ms, am I missing something?

Reply:
Yeah, right now we dont provide any specific result here.  The idea was that if there was some message that we needed to override the return we'd be able to specify a value for the outer window (the modal windows in this case) to return.  This code looks ok for now.  -- This is code that I expect to all be replaced by Island stuff long term.